packages/devtools_app/lib/src/timeline/flutter/event_details.dart

```diff
@@ -15,25 +15,39 @@ import '../timeline_model.dart';
 class EventDetails extends StatelessWidget {
   const EventDetails(this.selectedEvent);
 
+  static const instructions =
+      'Select an event from the Timeline to view details';
+  static const noEventSelected = '[No event selected]';
+
   final TimelineEvent selectedEvent;
 
   @override
   Widget build(BuildContext context) {
+    final textTheme = Theme.of(context).textTheme;
     return Column(
       crossAxisAlignment: CrossAxisAlignment.start,
       children: [
         Padding(
           padding: const EdgeInsets.only(top: 8.0),
           child: Text(
-            '${selectedEvent.name} - ${msText(selectedEvent.time.duration)}',
-            style: Theme.of(context).textTheme.title,
+            selectedEvent != null
+                ? '${selectedEvent.name} - ${msText(selectedEvent.time.duration)}'
+                : noEventSelected,
+            style: textTheme.title,
           ),
         ),
         const PaddedDivider.thin(),
         Expanded(
-          child: selectedEvent.isUiEvent
-              ? CpuProfilerView()
-              : EventSummary(selectedEvent),
+          child: selectedEvent != null
+              ? selectedEvent.isUiEvent
```

Comment:
This is hard to understand at a glance.  Can you refactor?

Reply:
done. used helpers